lib/tlscontext.c

```diff
@@ -697,6 +675,50 @@ tls_context_load_key_and_cert(TLSContext *self)
   return TLS_CONTEXT_OK;
 }
 
+static void _write_line_to_keylog_file_mutex(const char *file_path, const char *line)
+{
+  FILE *keylog_file;
+  gint ret_val;
+  static GStaticMutex mutex;
+  g_static_mutex_lock(&mutex);
+  keylog_file = fopen(file_path, "a");
```

Comment:
Opening and closing the file for every line might be problematic.
We could add `keylog_file` to `TLSContext`, next to the keylog file's lock.